core/macros/src/class.rs

```diff
@@ -0,0 +1,824 @@
+            },
+            is_static: false,
+        })
+    }
+}
+
+#[derive(Debug, Default)]
+struct Accessor {
+    getter: Option<Function>,
+    setter: Option<Function>,
+}
+
+impl Accessor {
+    fn set_getter(
+        &mut self,
+        name: String,
+        fn_: &mut ImplItemFn,
+        class_ty: &Type,
+    ) -> SpannedResult<()> {
+        let getter = Function::getter(name, fn_, class_ty)?;
+        if self.getter.is_some() {
+            error(fn_, "Getter for this property already declared.")
+        } else {
+            self.getter = Some(getter);
+            Ok(())
+        }
+    }
+
+    fn set_setter(
+        &mut self,
+        name: String,
+        fn_: &mut ImplItemFn,
+        class_ty: &Type,
+    ) -> SpannedResult<()> {
+        let setter = Function::setter(name, fn_, class_ty)?;
+        if self.setter.is_some() {
+            error(fn_, "Setter for this property already declared.")
+        } else {
+            self.setter = Some(setter);
+            Ok(())
+        }
+    }
+
+    fn body(&self) -> TokenStream2 {
+        let Some(name) = self
+            .getter
+            .as_ref()
+            .map_or_else(|| self.setter.as_ref().map(|s| &s.name), |g| Some(&g.name))
+        else {
+            return quote! {};
+        };
+        let getter = if let Some(getter) = self.getter.as_ref() {
+            let body = getter.body.clone();
+            quote! {
+                Some(
+                    boa_engine::NativeFunction::from_copy_closure( #body )
+                        .to_js_function(builder.context().realm())
+                )
+            }
+        } else {
+            quote! { None }
+        };
+        let setter = if let Some(setter) = self.setter.as_ref() {
+            let body = setter.body.clone();
+            quote! {
+                Some(
+                    boa_engine::NativeFunction::from_copy_closure( #body )
+                        .to_js_function(builder.context().realm())
+                )
+            }
+        } else {
+            quote! { None }
+        };
+
+        quote! {
+            {
+                let g = #getter;
+                let s = #setter;
+                builder.accessor(
+                    boa_engine::js_string!( #name ),
+                    g,
+                    s,
+                    boa_engine::property::Attribute::CONFIGURABLE
+                        | boa_engine::property::Attribute::NON_ENUMERABLE,
+                );
+            }
+        }
+    }
+}
+
+#[derive(Debug, Default)]
+enum RenameScheme {
+    #[default]
+    None,
+    CamelCase,
+}
+
+impl FromStr for RenameScheme {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.eq_ignore_ascii_case("none") {
+            Ok(Self::None)
+        } else if s.eq_ignore_ascii_case("camelcase") {
+            Ok(Self::CamelCase)
+        } else {
+            Err(format!("Invalid rename scheme: {s:?}"))
```

Comment:
Remark: Since they get turned into compile error messages, maybe also print the names that are acceptable.